### ISSUE TITLE
ALOAD 0 randomly being an issue

### ISSUE THREAD

**Issue by Anonymous-275** [Col-E/Recaf#236]
**Describe the bug**

Using the Assembler to view the bytecode sometimes I encounter ALOAD 0 being an issue (Varification failed on line : x) (Error at instruction x: Copying an uninitialized value should not occur!)

**To Reproduce**

1. load Main.class
2. Navigate to the methods tab (table mode)
3. Open the public static void main method with a string array as an argument
4. See error on line 955

**Screenshots**

![Capture](https://user-images.githubusercontent.com/36374260/88936761-ed323880-d28b-11ea-9535-005c305fa352.PNG)
![Capture1](https://user-images.githubusercontent.com/36374260/88936766-ef949280-d28b-11ea-8d38-1e20870794a2.PNG)

[Main.zip](https://github.com/Col-E/Recaf/files/5001891/Main.zip)

**Comment by Anonymous-275** [comment]
also worth mentioning other classes like this one having ALOAD being an issue in almost every method (even very small methods and mainly ALOAD 0 and ALOAD 1)
[CraftEventFactory.zip](https://github.com/Col-E/Recaf/files/5001945/CraftEventFactory.zip)

**Comment by Col-E** [comment]
It usually deals with obf, but smaller samples would definitely help me debug the problem better. I'll look at the sample soon.

**Comment by Anonymous-275** [comment]
The 2 zip files each contain 1 class that has that issue they are the smallest I could find yet if I do find even smaller classes I will definitely provide them

**Comment by Anonymous-275** [comment]
here i managed to replicate the issue with a class I just compiled 
here is the source code : 
```
public static byte[] sendRequest(String a,String b){
        System.out.println("Called with : " + a + " and " + b);
        return new byte[1];
}
```

here is the class:
[Test.zip](https://github.com/Col-E/Recaf/files/5002621/Test.zip)

**Comment by Col-E** [comment]
Woaaah that's a small sample. Looks like handling parameters is screwed up.

**Comment by Anonymous-275** [comment]
Yup exactly what just clicked into my head since all the others were using the argument at the same location so I tested it and my theory was right I hope it's enough to information to fix it since it's quite annoying not being able to edit it (even if I turn off verification it will literally break the method)

**Comment by Col-E** [comment]
I coulda sworn this wasn't an issue before. Must have been a regression in how parameters are handled in [SimAnalyzer](https://github.com/Col-E/SimAnalyzer)

**Comment by Anonymous-275** [comment]
Yeah it's rather odd since ALOAD 1 works here and I had encountered ALOAD 1 not working in other classes it also seems to always flags one ALOAD instruction per class

**Comment by Col-E** [comment]
That's actually because the assembler only reports one problem at a time since that's how basic exception handling works in Java.

**Comment by Anonymous-275** [comment]
interesting yup can confirm ALOAD 1 breaks in the same Test.class if i remove ALOAD 0 xD

**Comment by Col-E** [comment]
![image](https://user-images.githubusercontent.com/21371686/88995098-d30e4f80-d2b7-11ea-8234-00f2bf454c1b.png)


Uhh.... it works for me 😕 😖

**Comment by Col-E** [comment]
I'm marking this as _cant reproduce_ for the smaller samples since this doesn't appear without manual effort.

The manual effort breaking the assembler is still a bug, but it is not in that state by default so its lower priority IMO.

**Comment by Anonymous-275** [comment]
Yeah that's fine with the other classes but I am very confused now I am using the latest recaf didn't touch any settings yet when I decompile this it gives me errors could it be related to the java version it's loaded in?

**Comment by Anonymous-275** [comment]
ok that is very funny when i do it it's 0 and 1 when I rename the parameters to a and b but leave it as ALOAD 0 and ALOAD 1 it works just fine idk how at this point you have a different outcome when on my end it's very consistent with the test.class :/

**Comment by Anonymous-275** [comment]
figured it out turns out somehow having a very old version of recaf (before the UI revap yes that old) and upgrading to the latest one broke it when I deleted the appdata folder called recaf it now works beautifully

**Comment by Anonymous-275** [comment]
found the culprit the settings folder here is how it managed to break by default 
replacing the config with this makes it break by default
[config.zip](https://github.com/Col-E/Recaf/files/5006607/config.zip)

**Comment by Anonymous-275** [comment]
ok even better i found the config choice that broke it
![Capture](https://user-images.githubusercontent.com/36374260/89028220-97fe3180-d334-11ea-85b2-687533127b4a.PNG)

**Comment by Anonymous-275** [comment]
And yes you are correct the default option for that is false but giving the fact that i had to use it in some cases to strip debug information then I would say it's a somewhat and important/annoying on that note I am a very old user of recaf (since 2016 or 2017) and i love it ever since keep up the amazing work! :D

**Comment by Col-E** [comment]
Strip debug shouldn't affect anything other than decompiling 😬 

That's a bug. 

Idea: Split it into two, so you can opt to strip it for one, the other, both, or neither

**Comment by Anonymous-275** [comment]
Yeah that's what I thought since there is a tab dedicated in the settings either way I did manage to steer my way through the complications and make it work, and since I did there is no real ruch here tho it would be interesting to know if there is any ETA :D

**Comment by Col-E** [comment]
Should be a simple fix. Depends on how tired I am after work how fast I get to it.

**Comment by Anonymous-275** [comment]
No problem! Keep up the amazing work :D

**Comment by Col-E** [comment]
Resolved in `2.4.0` which will be pushed out soon
